ksql-parser/src/main/java/io/confluent/ksql/parser/tree/CreateSourceProperties.java

```diff
@@ -62,326 +176,124 @@
         "HOPPING", () -> WindowedSerdes.timeWindowedSerdeFrom(String.class)
       );
 
-  // required
-  private final Property<String> kafkaTopic;
-  private final Property<Format> valueFormat;
-
-  // optional
-  private final Property<String> key;
-  private final Property<String> windowType;
-  private final Property<String> timestampName;
-  private final Property<String> timestampFormat;
-  private final Property<String> avroSchemaName;
-  private final Property<Integer> avroSchemaId;
-  private final Property<Integer> partitions;
-  private final Property<Short> replicas;
-  private final Property<Boolean> wrapSingleValues;
-
-  public CreateSourceProperties(final Map<String, Literal> original) {
-    final Map<String, Literal> properties = original
-        .entrySet()
-        .stream()
-        .collect(Collectors.toMap(entry -> entry.getKey().toUpperCase(), Entry::getValue));
-
-    properties.keySet()
-        .stream()
-        .filter(cfg -> !VALID_PROPERTIES.contains(cfg))
-        .findFirst()
-        .ifPresent(config -> {
-          throw new KsqlException("Invalid config variable in the WITH clause: " + config);
-        });
-
-    valueFormat = Property.from(
-        DdlConfig.VALUE_FORMAT_PROPERTY, properties, val -> Format.of(val.toString()));
-    if (!valueFormat.isPresent()) {
-      throw new KsqlException("Topic format(VALUE_FORMAT) should be set in WITH clause.");
-    }
+  private final ImmutableMap<String, Literal> originalLiterals;
 
-    kafkaTopic = Property.from(
-        DdlConfig.KAFKA_TOPIC_NAME_PROPERTY,
-        properties,
-        ((Function<Object, String>) Object::toString).andThen(StringUtil::cleanQuotes));
```

Comment:
i see that unit tests are passing, but I'm wondering how you got it to clean quotes...

Reply:
I'm using `Literal.getValue`...

Reply:
🤦‍♂ got it